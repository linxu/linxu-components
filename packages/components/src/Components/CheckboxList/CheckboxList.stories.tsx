import React from "react";
import { Story } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import { getTheme } from "../../theme/getTheme";
import {
  CheckboxListProps,
  ExomeCheckboxList,
} from "./CheckboxList";

export default {
  title: "Components/CheckboxList",
  component: ExomeCheckboxList,
};

export const Minimal: Story<CheckboxListProps> = (props: CheckboxListProps) => (
  <ThemeProvider theme={getTheme()}>
    <ExomeCheckboxList {...props} />
  </ThemeProvider>
);

Minimal.args = {
  data: [
    { id: 23424, name: "Carrots", summary: "USD30,00" },
    { id: 43534, name: "Potatoes", summary: "USD45,00" },
  ],
  nameKey: "name",
  summaryKey: "summary",
  // eslint-disable-next-line no-unused-vars
  onChecked: (_: number[]) => {},
};

export const WithoutData: Story<CheckboxListProps> = (
  props: CheckboxListProps
) => (
  <ThemeProvider theme={getTheme()}>
    <ExomeCheckboxList {...props} />
  </ThemeProvider>
);

WithoutData.args = {};

export const DefaultKeys: Story<CheckboxListProps> = (
  props: CheckboxListProps
) => (
  <ThemeProvider theme={getTheme()}>
    <ExomeCheckboxList {...props} />
  </ThemeProvider>
);

DefaultKeys.args = {
  data: [
    { id: 23424, name: "Carrots", summary: "USD30,00" },
    { id: 43534, name: "Potatoes", summary: "USD45,00" },
  ],
};
