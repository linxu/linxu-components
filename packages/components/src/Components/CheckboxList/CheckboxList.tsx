import React, { useState } from "react";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { Checkbox } from "../Checkbox/Checkbox";
import { Text } from "../Text/Text";

type Data = { id: number } & any;

export interface CheckboxListProps {
  /** data source, e.g.: {name: "test", summary: 1} */
  data: Data[];
  /** Callback fired when the ScCheckbox is changed */
  // eslint-disable-next-line no-unused-vars
  onChecked: (_: number[]) => void;
  /** name key from data */
  nameKey: string;
  /** summary key from data */
  summaryKey: string;
}

// CheckboxList Component, it is a simple Checkbox List.
export const CheckboxList = ({
  data = [],
  nameKey = "name",
  summaryKey = "summary",
  onChecked,
}: CheckboxListProps) => {
  const [checked, setChecked] = useState<number[]>([]);

  return (
    <List>
      {data.map((item, index) => (
        <ListItem
          key={item.id}
          data-testid={`checkbox-list-item-${index + 1}`}
          dense
          button
          divider
          onClick={() => {
            const currentIndex = checked.indexOf(item.id);
            const newChecked = [...checked];
            if (currentIndex === -1) {
              newChecked.push(item.id);
            } else {
              newChecked.splice(currentIndex, 1);
            }
            setChecked(newChecked);
            if (onChecked) {
              onChecked(newChecked);
            }
          }}
          style={{ paddingLeft: 0 }}
        >
          <ListItemIcon style={{ minWidth: 0 }}>
            <Checkbox checked={checked.indexOf(item.id) !== -1} />
          </ListItemIcon>
          <ListItemText>
            <Text fontSize="medium">{item[nameKey]}</Text>
          </ListItemText>
          {item[summaryKey] && (
            <Text fontSize="medium">{item[summaryKey]}</Text>
          )}
        </ListItem>
      ))}
    </List>
  );
};

export const ExomeCheckboxList = CheckboxList;
