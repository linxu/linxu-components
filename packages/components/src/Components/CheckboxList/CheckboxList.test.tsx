import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

import {
  DefaultKeys,
  Minimal,
} from "./CheckboxList.stories";
import { CheckboxListProps } from "./CheckboxList";

it("renders", async () => {
  const tree = render(<Minimal {...(Minimal.args as CheckboxListProps)} />);
  const item = await screen.findByTestId("checkbox-list-item-1");
  await fireEvent.click(item);
  await fireEvent.click(item);
  expect(tree).toMatchSnapshot();
});
it("renders", async () => {
  const tree = render(
    <DefaultKeys {...(DefaultKeys.args as CheckboxListProps)} />
  );
  const item = await screen.findByTestId("checkbox-list-item-1");
  await fireEvent.click(item);
  expect(tree).toMatchSnapshot();
});
