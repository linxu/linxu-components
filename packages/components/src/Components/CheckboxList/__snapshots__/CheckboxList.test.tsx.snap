// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          data-testid="checkbox-list-item-1"
          role="button"
          style="padding-left: 0px;"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
            style="min-width: 0;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-4"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  class="sc-bdfBQB kleLNI"
                  color="secondary"
                  fill="#21222F"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="white"
                    height="13.3333"
                    width="13.3333"
                    x="1.33398"
                    y="1.33398"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-dense"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
            >
              <p
                class="sc-dlfnuX cigaJp"
                font-size="medium"
              >
                Carrots
              </p>
            </span>
          </div>
          <p
            class="sc-dlfnuX cigaJp"
            font-size="medium"
          >
            USD30,00
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          data-testid="checkbox-list-item-2"
          role="button"
          style="padding-left: 0px;"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
            style="min-width: 0;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-4"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  class="sc-bdfBQB kleLNI"
                  color="secondary"
                  fill="#21222F"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="white"
                    height="13.3333"
                    width="13.3333"
                    x="1.33398"
                    y="1.33398"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-dense"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
            >
              <p
                class="sc-dlfnuX cigaJp"
                font-size="medium"
              >
                Potatoes
              </p>
            </span>
          </div>
          <p
            class="sc-dlfnuX cigaJp"
            font-size="medium"
          >
            USD45,00
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        data-testid="checkbox-list-item-1"
        role="button"
        style="padding-left: 0px;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
          style="min-width: 0;"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                class="sc-bdfBQB kleLNI"
                color="secondary"
                fill="#21222F"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="white"
                  height="13.3333"
                  width="13.3333"
                  x="1.33398"
                  y="1.33398"
                />
                <path
                  clip-rule="evenodd"
                  d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            <p
              class="sc-dlfnuX cigaJp"
              font-size="medium"
            >
              Carrots
            </p>
          </span>
        </div>
        <p
          class="sc-dlfnuX cigaJp"
          font-size="medium"
        >
          USD30,00
        </p>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        data-testid="checkbox-list-item-2"
        role="button"
        style="padding-left: 0px;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
          style="min-width: 0;"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                class="sc-bdfBQB kleLNI"
                color="secondary"
                fill="#21222F"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="white"
                  height="13.3333"
                  width="13.3333"
                  x="1.33398"
                  y="1.33398"
                />
                <path
                  clip-rule="evenodd"
                  d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            <p
              class="sc-dlfnuX cigaJp"
              font-size="medium"
            >
              Potatoes
            </p>
          </span>
        </div>
        <p
          class="sc-dlfnuX cigaJp"
          font-size="medium"
        >
          USD45,00
        </p>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          data-testid="checkbox-list-item-1"
          role="button"
          style="padding-left: 0px;"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
            style="min-width: 0;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  class="sc-bdfBQB kleLNI"
                  color="success"
                  fill="#47BA67"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="white"
                    height="12"
                    width="12"
                    x="2"
                    y="2"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM3.66602 8.66803L6.5436 11.666L12.3327 5.66298L11.3758 4.66602L6.5436 9.67209L4.62294 7.67106L3.66602 8.66803Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-dense"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
            >
              <p
                class="sc-dlfnuX cigaJp"
                font-size="medium"
              >
                Carrots
              </p>
            </span>
          </div>
          <p
            class="sc-dlfnuX cigaJp"
            font-size="medium"
          >
            USD30,00
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          data-testid="checkbox-list-item-2"
          role="button"
          style="padding-left: 0px;"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
            style="min-width: 0;"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-8"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  class="sc-bdfBQB kleLNI"
                  color="secondary"
                  fill="#21222F"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="white"
                    height="13.3333"
                    width="13.3333"
                    x="1.33398"
                    y="1.33398"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-dense"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
            >
              <p
                class="sc-dlfnuX cigaJp"
                font-size="medium"
              >
                Potatoes
              </p>
            </span>
          </div>
          <p
            class="sc-dlfnuX cigaJp"
            font-size="medium"
          >
            USD45,00
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        data-testid="checkbox-list-item-1"
        role="button"
        style="padding-left: 0px;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
          style="min-width: 0;"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-8"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                class="sc-bdfBQB kleLNI"
                color="success"
                fill="#47BA67"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="white"
                  height="12"
                  width="12"
                  x="2"
                  y="2"
                />
                <path
                  clip-rule="evenodd"
                  d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM3.66602 8.66803L6.5436 11.666L12.3327 5.66298L11.3758 4.66602L6.5436 9.67209L4.62294 7.67106L3.66602 8.66803Z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            <p
              class="sc-dlfnuX cigaJp"
              font-size="medium"
            >
              Carrots
            </p>
          </span>
        </div>
        <p
          class="sc-dlfnuX cigaJp"
          font-size="medium"
        >
          USD30,00
        </p>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
        data-testid="checkbox-list-item-2"
        role="button"
        style="padding-left: 0px;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
          style="min-width: 0;"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary sc-gsTEea MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-8"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                class="sc-bdfBQB kleLNI"
                color="secondary"
                fill="#21222F"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="white"
                  height="13.3333"
                  width="13.3333"
                  x="1.33398"
                  y="1.33398"
                />
                <path
                  clip-rule="evenodd"
                  d="M2.66602 0.666016C1.56145 0.666016 0.666016 1.56145 0.666016 2.66602V13.3327C0.666016 14.4373 1.56145 15.3327 2.66602 15.3327H13.3327C14.4373 15.3327 15.3327 14.4372 15.3327 13.3327V2.66602C15.3327 1.56145 14.4372 0.666016 13.3327 0.666016H2.66602ZM2.99935 1.99935C2.44706 1.99935 1.99935 2.44706 1.99935 2.99935V12.9993C1.99935 13.5516 2.44706 13.9993 2.99935 13.9993H12.9993C13.5516 13.9993 13.9993 13.5516 13.9993 12.9993V2.99935C13.9993 2.44706 13.5516 1.99935 12.9993 1.99935H2.99935Z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            <p
              class="sc-dlfnuX cigaJp"
              font-size="medium"
            >
              Potatoes
            </p>
          </span>
        </div>
        <p
          class="sc-dlfnuX cigaJp"
          font-size="medium"
        >
          USD45,00
        </p>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
